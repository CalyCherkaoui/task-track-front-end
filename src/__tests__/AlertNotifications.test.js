import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import AlertsNotifications from '../component/AlertNotifications';

describe(
  'Renders correctly The navigation bar component',
  () => {
    let container;
    const message = 'Test Notification';
    beforeEach(() => {
      container = shallow(
        <Provider store={store}>
          <Router>
            <AlertsNotifications message={message} />
          </Router>
        </Provider>,
      );
    });

    it(
      'Contains Alert\'s message',
      () => {
        expect(container.find('Test Notification')).toBeTruthy();
      },
    );

    it(
      'Matches AlertNotifications snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <AlertsNotifications message={message} />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
