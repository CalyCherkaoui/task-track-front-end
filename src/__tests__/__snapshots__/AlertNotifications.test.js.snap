// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Alerts notifications component Matches AlertNotifications snapshot 1`] = `
<div>
  <div
    id="success_notif"
    style={
      Object {
        "color": "green",
        "display": "none",
      }
    }
  >
    <div
      className="fade alert alert-success show"
      role="alert"
    >
      Test Notification
    </div>
  </div>
  <div
    id="errors_notif"
    style={
      Object {
        "color": "red",
        "display": "none",
      }
    }
  >
    <div
      className="fade alert alert-danger show"
      role="alert"
    >
      Oops! Something went wrong! Try again!
    </div>
  </div>
</div>
`;
